sdk/core/Azure.Core/src/Shared/NextLinkOperationImplementation.cs

```diff
@@ -70,6 +72,41 @@ internal class NextLinkOperationImplementation : IOperation
             return new OperationToOperationOfT<T>(operationSource, operation);
         }
 
+        public static IOperation Create(
+            HttpPipeline pipeline,
+            RehydrationToken? rehydrationToken,
+            string? apiVersionOverride = null)
+        {
+            Argument.AssertNotNull(rehydrationToken, nameof(rehydrationToken));
+
+            var lroDetails = ModelReaderWriter.Write(rehydrationToken!, new ModelReaderWriterOptions("J")).ToObjectFromJson<Dictionary<string, string>>();
+            if (!Uri.TryCreate(lroDetails["initialUri"], UriKind.Absolute, out var startRequestUri))
+                throw new InvalidOperationException("Invalid initial URI");
+            if (!lroDetails.TryGetValue("nextRequestUri", out var nextRequestUri))
+                throw new InvalidOperationException("Invalid next request URI");
+            RequestMethod requestMethod = new RequestMethod(lroDetails["requestMethod"]);
+            string lastKnownLocation = lroDetails["lastKnownLocation"];
+            if (!Enum.TryParse(lroDetails["finalStateVia"], out OperationFinalStateVia finalStateVia))
+                finalStateVia = OperationFinalStateVia.Location;
+            string? apiVersionStr = apiVersionOverride ?? (TryGetApiVersion(startRequestUri, out ReadOnlySpan<char> apiVersion) ? apiVersion.ToString() : null);
+            if (!Enum.TryParse(lroDetails["headerSource"], out HeaderSource headerSource))
```

Comment:
Do we trust this?   If we don't have an `Enum.IsDefined` check backing this up, we could very well accept an invalid value.  For example:

```csharp
Enum.TryParse("9999", out OperationFinalStateVia finalStateVia)
```
This will return `true` and `finalStateVia` will equal 9999, despite it not being a valid enum member.

Reply:
updated